python/eos/analysis_file_description.py

```diff
@@ -239,6 +240,9 @@ def __post_init__(self):
     @classmethod
     def from_dict(cls, **kwargs):
         _kwargs = _copy.deepcopy(kwargs)
+        if "default_arguments" in kwargs:
```

Comment:
Hm, I would prefer to apply this logic in the execution, i.e., the ``run`` task, not the description.

Reply:
By doing it here, the `__post_init__` step of `TaskComponent` gives an error if you have `default_arguments` that don't apply to the tasks. If you want to do it in `run`, you have to do some checks there, else the task will fail in the case of wrong arguments.

Reply:
At the meeting, agreed to move this logic into ``AnalysisFile.validate()``.

Reply:
How do you feel about this new version? Default arguments are kept separate, but checked in `AnalysisFile.validate`, and applied in the `run` task.
Note I've also altered how `default_arguments` are specified - they are now per task, as this keeps the mapping between CLI and internal names clean, and allows you to have multiple different tasks specified in a single step.

Reply:
Note I am coming around to thinking we should have a specific task component dataclasses, i.e. a `SampleNestedTaskConfiguration`, ...  (similar to what is done for Priors), and so enforce more of the structure via the AnalysisFileDescription code. To be discussed...